src/main/java/seedu/address/model/tier/Tier.java

```diff
@@ -52,19 +56,30 @@ public boolean equals(Object other) {
         }
 
         Tier otherTier = (Tier) other;
-        return tagName.equals(otherTier.tagName);
+        return tierName.equals(otherTier.tierName);
     }
 
     @Override
     public int hashCode() {
-        return tagName.hashCode();
+        return tierName.hashCode();
     }
 
     /**
      * Format state as text for viewing.
      */
     public String toString() {
-        return '[' + tagName.toString() + ']';
+        return '[' + tierName.toString() + ']';
+    }
+
+    /**
+     * Format Tier for parsing/decoding. As TierEnum.NA is only for internal use,
+     * if tierName equals TierEnum.NA, an empty string is returned
+     */
+    public String toDecodeString() {
+        if (tierName.equals(TierEnum.NA)) {
+            return "";
+        }
+        return tierName.toString();
     }
```

Comment:
Could `toDisplayString()` be a more descriptive name for the method? Decode seems a little verbose here to me.